DXRModules/DeusEx/Classes/DXRBingoCampaign.uc

```diff
@@ -0,0 +1,257 @@
+class DXRBingoCampaign extends DXRActorsBase transient;
+
+function PreFirstEntry()
+{
+    local DXREvents events;
+    local BlackHelicopter jock;
+    local Switch1 button;
+    local PayPhone invisibleWall;
+    local #var(prefix)FlagTrigger ft;
+
+    if (!dxr.flags.IsBingoCampaignMode()) return;
+    Super.PreFirstEntry();
+
+    foreach AllActors(class'DXREvents', events) {
+        switch (dxr.localURL) {
+            case "02_NYC_BATTERYPARK":
+                dxr.flags.settings.starting_map = 20;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "03_NYC_UNATCOISLAND":
+                dxr.flags.settings.starting_map = 30;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "03_NYC_AIRFIELD":
+                foreach AllActors(class'BlackHelicopter', jock) {
+                    AddBingoEventBlocker(jock, 'DXRando_Mission03_BingoCompleted');
+                    break;
+                }
+                break;
+            case "04_NYC_UNATCOISLAND":
+                dxr.flags.settings.starting_map = 40;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "05_NYC_UNATCOMJ12LAB":
+                if (!dxr.flagbase.GetBool('DXRando_Mission04_BingoCompleted')) {
+                    SetTimer(1.0, true);
+                }
+                dxr.flags.settings.starting_map = 50;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "06_HONGKONG_HELIBASE":
+                dxr.flags.settings.starting_map = 60;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "08_NYC_STREET":
+                dxr.flags.settings.starting_map = 80;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "09_NYC_Dockyard":
+                dxr.flags.settings.starting_map = 90;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "10_PARIS_CATACOMBS":
+                dxr.flags.settings.starting_map = 100;
+                events.CreateBingoBoard();
+                ClearDataVaultImages();
+                break;
+            case "10_PARIS_CHATEAU":
+                if (class'DXRMapVariants'.static.IsVanillaMaps(player())) {
+                    invisibleWall = PayPhone(Spawnm(class'PayPhone',, 'invisible_wall', vect(-999.7, -4484.5, -174.4)));
```

Comment:
I'm happy to learn that there's a better way to accomplish it, but I certainly feel that the wall should be invisible. There's no lore purpose for you to be unable to continue. It's bingo magically blocking your progress.

A hover hint isn't a bad idea though.

Reply:
how does this look? it doesn't need to be invisible, it can be unbreakable/unmovable big metal crates or something, or barricades, or just make it so you can't open the doors?

Reply:
I like it as an invisible wall. You're not being blocked, conceptually, by something in-game. JC is just refusing to go forward because he hasn't finished his bingo yet.

Reply:
Players need feedback that that's the case though.  Visual has the added advantage of working for non-english speakers or those who don't/can't read (aka DXRando Players)

Reply:
It not about reading. They just have to remember that they're in a Page Machine game and they should be able to figure it out.

Reply:
It's just not good game design to not give any player feedback.  An invisible wall in the path of progression is just going to confuse people.

Reply:
A force field texture is my next favorite idea.

Reply:
invisible walls are never good, that's like such a common complaint about games in general

and "remembering" only works if they're already familiar, remember we've seen players attempt WaltonWare and Halloween Mode and stuff without even remembering the basics, we've seen people not know things like needing to talk to Filben before Dowd appears, or how many weld points there are, or that you can use the computer to destroy the generator. There was a time when you were playing WaltonWare and you didn't know where missions officially started and ended.

Reply:
you're also assuming that players will even know what the game mode is before playing it lol

Reply:
> invisible walls are never good, that's like such a common complaint about games in general

I feel like this is a special case. The immersion breaking here is justified by the mode you've chosen, which is inherently unimmersive.

Reply:
If the immersion is already broken, then why not make it clear?